src/MfaVerifyController.js

```diff
@@ -100,6 +115,14 @@ function (Okta, Checkbox, BaseLoginController, TOTPForm, YubikeyForm, SecurityQu
           this.options.appState.get('isMfaRequired')) {
         return true;
       }
+      if (this.options.appState.get('isSuccessResponse')) {
```

Comment:
Yeah, the general idea is that the backend controls the auth flow - in the case where mfa is required, we don't go from primary auth -> password_expired, we go from primaryauth -> mfa -> password_expired.

That's one of the reasons why the controllers are set up as they are - they're in charge of getting to the next state, but they never know what the next state is (and why we'll never see logic like here where we say something like "if the state is success, do this").